In [1]:
import pandas as pd
all_data = pd.read_csv('global_temp.csv')
recent = pd.read_csv('recent_temp.csv')

Verify using a hypothesis test if the temperature in New Delhi has gone down in the recent years.

The file `global_temp.csv`, contains the average temperature for each city for a period starting from 2018 to 2020.

The file `recent_temp.csv`, contains the temperature data for the year 2021.

In [2]:
all_data.head(2)

city   avg_temp
0    Sukhumi  15.116536
1  Mariehamn   7.336081

In [3]:
recent.head(2)

date   country     city   Latitude  Longitude  tavg  tmin  tmax  \
0  2021-01-01  Abkhazia  Sukhumi  43.001525  41.023415   8.6   5.4  14.9   
1  2021-01-02  Abkhazia  Sukhumi  43.001525  41.023415   8.5   4.0  13.7   

   wdir  wspd    pres  
0  81.0   9.4  1019.2  
1  47.0   7.7  1024.8

In [4]:
all_data[all_data['city']=='New Delhi']

city   avg_temp
73  New Delhi  26.133759

In [10]:
pop_mean = all_data[all_data['city']=='New Delhi']['avg_temp'].values

In [5]:
recent[recent['city']=='New Delhi']['tavg'].mean()

25.87123287671233

In [6]:
sample_avg = recent[recent['city']=='New Delhi']['tavg'].mean()

In [7]:
sample_stdev = recent[recent['city']=='New Delhi']['tavg'].std()

In [8]:
sample_size = recent[recent['city']=='New Delhi']['tavg'].shape[0]

H0: Avg Temp = 26.133759

Ha: Avg Temp < 26.133759

p-value = Pr(Temp<=25..87123287671233)

Standard Deviation of Sampling Distribution = 7.044814932408316/sqrt(365)

In [9]:
import math
import scipy.stats
stdev_samp = sample_stdev/math.sqrt(sample_size)

In [11]:
p_value = scipy.stats.norm(pop_mean,stdev_samp).cdf(sample_avg)

In [12]:
p_value

array([0.23824783])

`Use the data for city Ottawa and verify if the sample deviation is actually statistically significant.`

In [16]:
recent[recent['city']=='Ottawa']['tavg'].mean()

8.093150684931507

In [13]:
sample_avg = recent[recent['city']=='Ottawa']['tavg'].mean()

In [14]:
sample_stdev = recent[recent['city']=='Ottawa']['tavg'].std()

In [15]:
sample_size = recent[recent['city']=='Ottawa']['tavg'].shape[0]

In [18]:
all_data[all_data['city']=='Ottawa']['avg_temp']

25    6.79854
Name: avg_temp, dtype: float64

In [16]:
pop_mean = all_data[all_data['city']=='Ottawa']['avg_temp'].values

```
H0: Temp=6.79854
H1: Temp>6.79854

p-value = Pr(Temp>=8.093150684931507) = 1-Pr(Temp<=8.093150684931507)

Stdev Sampling Distribution = 11.507309296430472/sqrt(365)
```

In [17]:
stdev_samp = sample_stdev/math.sqrt(sample_size)
p_value = 1 - scipy.stats.norm(pop_mean,stdev_samp).cdf(sample_avg)

In [18]:
p_value

array([0.01580242])